cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonLanguageFrontend.kt

```diff
@@ -76,9 +83,26 @@ class PythonLanguageFrontend(language: Language<PythonLanguageFrontend>, ctx: Tr
             it.set("content", fileContent)
             it.set("filename", file.absolutePath)
             it.exec("import ast")
+            it.exec("import sys")
             it.exec("parsed = ast.parse(content, filename=filename, type_comments=True)")
 
             val pyAST = it.getValue("parsed") as PyObject
+
+            // Populate system information
+            SysInfo.platform = it.getValue("sys.platform") as? String
```

Comment:
I'm not sure at all if this is really what we want. This checks the platform and python version which we use to analyze/parse the code which can be completely different from the environment which is used to execute the code. This is likely to introduce problems.

I'd prefer to pass such information to the CPG via arguments, configuration files or similar (as it's done below I guess?) which represent the "execution environment" of a certain program (if such information is available). IMHO it's better to have no information here instead of wrong assumptions.

Furthermore, I'd expect to find the SysInfo object somewhere in the translation result since these are some assumptions which may be useful for further analysis.

Reply:
Agree, fetching this from here was my initial approach and then I added the "environment" below, it probably makes sense to only have the environment symbols here.